org/newEigenBeta.ts

```diff
@@ -0,0 +1,117 @@
+import { Create } from "github-webhook-event-types"
+import { danger } from "danger"
+
+import * as semverSort from "semver-sort"
+import * as JiraApi from "jira-client"
+import { flatten } from "lodash"
+
+import * as IssueJSON from "../fixtures/jira_issue_example.json"
+type Issue = typeof IssueJSON
+
+const companyPrefix = "artsyproduct"
+
+// Implementation borrowed from https://github.com/danger/peril-settings/blob/b306662628f1a4c5c4ebc5f0531e15561fe60fa5/org/new_tag.ts#L12-L64
+export default async (create: Create) => {
+  if (create.ref_type !== "tag") {
+    return console.log("Skipping because it's not a tag")
+  }
+  if (create.ref.endsWith("-submission")) {
+    return console.log("Skipping because it's not a beta")
+  }
+
+  const api = danger.github.api
+
+  const tag = create.ref
+  const thisRepo = { owner: create.repository.owner.login, repo: create.repository.name }
+
+  // Grab all tags, should be latest ~20
+  const allTagsResponse = await api.repos.getTags(thisRepo)
+  const allTags: Tag[] = allTagsResponse.data
+
+  // Sort the tags in semver, so we can know specifically what range to
+  // work out the commits
+  const semverStrings: string[] = semverSort.desc(allTags.map(tag => tag.name))
+  const versionIndex = semverStrings.findIndex(version => version === tag)
+  const releaseMinusOne = semverStrings[versionIndex + 1]
+
+  // Bail if we can't find a release
+  if (!releaseMinusOne) {
+    return console.log(`Couldn't locate release for ${tag} tag.`)
+  }
+  // Ask for the commits
+  const compareResults = await api.repos.compareCommits({ ...thisRepo, base: releaseMinusOne, head: tag })
+  compareResults.data
+  const compareData: CompareResults = compareResults.data
+
+  // Pull out all the GH crafted merge commits on a repo
+  const numberExtractor = /Merge pull request #(\d*)/
+  const commitMessages = compareData.commits.map(c => c.commit.message)
+  const prMerges = commitMessages.filter(message => message && message.startsWith("Merge pull request #"))
+
+  // This is now a number array of PR ids
+  // e.g. [ 930, 934, 937, 932, 938 ]
+  const prs = prMerges
+    .map(msg => msg.match(numberExtractor) && msg.match(numberExtractor)![1])
+    .filter(pr => pr)
+    .map((pr: any) => parseInt(pr))
```

Comment:
I'd like to offer a refactor here - take or leave any or all of this!! ❤️ 

```typescript
const prs = prMerges.map(message => {
  const matches = message.match(numberExtractor) || []
  const capture = matches[1]
  return parseInt(capture)
}).filter(Boolean)
```

Reply:
This is great – I obviously trusted you that it works but I took a minute to break down exactly _why_ it works at each line, and it's quite elegant!

Reply:
Yay!! I have to admit that I did not actually write that locally and test it, so I probably should have had a disclaimer to that effect! 😝 

And nice job with ac791e2 - that default export reads really nice!! 😎